### DISCUSSION TITLE
What issues to get for the next release?

### DISCUSSION THREAD

**Discussion by oleg68** [GrandOrgue/grandorgue#715] [Developers]
@larspalo @nickappleton @rousseldenis 

Lets plan the next release.

We have already a lot of issues. What of them sould we do at first?

**Comment by midimusic** [comment]
Give each organ a defined default input configuration (rather than doing nothing until configured). This configuration should match the default output configuration which already exists.

Get rid of the undocumented sysex events at the start of GO midi files. The midi stop events and channel numbers should match the defaults above.

This would make playing the organ from midi files (often originally made on another organ) much easier.

csw900

**Reply by larspalo** [reply]
I think any new requests belong elsewhere! Create an issue - or rather maybe not for the first issue unless you refine it from before. The first request is an old ghost coming back to haunt us, you have already gotten an answer why this is not feasible in the way you suggested earlier. If you have anything new to suggest in the matter then please open that as an issue too.

**Reply by midimusic** [reply]
@lars:

The reason this ghost is still haunting is because it it the only **major** defect GO has in comparison with "other" organs. I do not know how I suggested it before but it is actually very easy to do because all the difficult code for generating and using the sysex is already present and correct. All you have to do is run this code immediately after the organ has loaded. The organ will then play standard midi files and will no longer need the sysex. There will be no compatibility issue because existing midi files will still work. It is optional whether you write the sysex to new midi files.
You can also optionally add another midi input to the organ (as other organs do) which uses this format. This format is already output from the midi recorder output.
I will give you a month or two to sort out your existing issues and then formally raise it as an issue as you suggested.
csw900

**Reply by larspalo** [reply]
You're unfortunately not really reading my reply in reference to what you actually wrote before. The ghost I referenced is your suggested initial MIDI configuration as was your first sentence above. Just to jog your memory - GO is a VPO not a MIDI player - we for instance have to accept that organ keyboard layouts won't necessarily adhere to any AGO standard, nor will stop naming or order of their appearance. There already exist a basic initial configuration possibility for the user - of course this is oriented mostly towards adapting new organs a bit faster to the console where you play from, because that's what GO really mostly is intended for. If you have suggestions as to how it could be improved, by all means create an issue for it. Just remember, there's no fixed universal standard as to exactly what MIDI messages should trigger what events in GO - the flexibility is there to adapt to whatever the user has available, that also comes with the price of the user needing to actually configure it.

**Reply by midimusic** [reply]
@larspalo 

Unfortunately you have not properly read and understood what I am **NOW** saying. Whatever I said before is history. Whether you like it or not GO **IS** an organ midi file maker and player any many users make use of this feature. Also many are frustrated by its poor design - which make is hard to use.

I am **NOT** suggesting any changes to the organ's existing midi input and setup procedure but only suggesting that GO should set up its midi file recorder/player after the organ is loaded rather than including the setup in every midi file.

PS My first sentence "Give each organ a defined default input configuration (rather than doing nothing until configured)" may have misled you. The changes I am suggesting will not affect the organ's normal midi input at all. The changes will affect only the midi file player's input to the organ.

csw900

**Comment by ajvdw** [comment]
Make the Midi Subsystems/midi devices selectable (related , extension to #703, #705), comparable with Audio (RtMidi, Alsa and Jack?)...

**Comment by rousseldenis** [comment]
> @larspalo @nickappleton @rousseldenis
> 
> Lets plan the next release.
> 
> We have already a lot of issues. What of them sould we do at first?

@oleg68 I like tags on issues and PR's, so, the first thing is to do a first triage affecting tags on them (I think we can add some - not too much - by categories).

**Comment by larspalo** [comment]
First focus should be on the obvious bugs, I think. Next are all the build related issues, both automatic and manual, and a revision on all the documentation both the build related and author related, not to mention cleaning up and updating all the copyright headers (or supply them where they don't exist).

**Reply by rousseldenis** [reply]
> First focus should be on the obvious bugs, I think. Next are all the build related issues, both automatic and manual, and a revision on all the documentation both the build related and author related, not to mention cleaning up and updating all the copyright headers (or supply them where they don't exist).

I agree. We can focus on simple things first, improving on the same time our flows.

**Comment by oleg68** [comment]
I've made a list.

@larspalo @rousseldenis Please do not put issues in another comments. Instead edit the same comment and add issues here.

- [x] #27 @oleg68 
- [x] #712 @larspalo 
- [x] #716 @rousseldenis 
- [x] #14 @oleg68
- [x] #713 @larspalo 
- [x] #683 @oleg68 

Please add here only links to issues. Add comment in your own comments

**Comment by oleg68** [comment]


**Reply by rousseldenis** [reply]
?

**Comment by oleg68** [comment]
@larspalo @rousseldenis Do you agree with the list above?

**Reply by larspalo** [reply]
I think your list is very good and sensible. We might add the versioning issue too to that list as that's closely related to the release numbering. Then it's just to decide who will work on what so that we won't do unnecessary double work.

**Comment by oleg68** [comment]
> Then it's just to decide who will work on what so that we won't do unnecessary double work.

#27 we all work on discussing. After we decide something I'll put the corresponding tag the the repo
#683 I am already in progress, so I think I should continue. Unfortunally we cann't use this ubuntu package untill we switch all other dependencies from opensuse to ubuntu, so the windows build in this release will the old.

**Comment by larspalo** [comment]
I've created a PR for #713 and will see later what I can do with #712. But if someone else fixes it before me I won't mind...

**Reply by oleg68** [reply]
OK, I've marked in the list

**Comment by oleg68** [comment]
@larspalo  @rousseldenis 

> Then it's just to decide who will work on what so that we won't do unnecessary double work.

For avoiding double work please assign the issues to youself if you are going to do them.

**Reply by rousseldenis** [reply]
Of course, that's what I did here https://github.com/GrandOrgue/grandorgue/issues/716

**Comment by oleg68** [comment]
Seems the copyright issue is the most complex. It is the only one rest for the relese.
@rousseldenis may we help you?

**Reply by rousseldenis** [reply]
I gave leads on how to do it.

As I don't want to be the only one to decide as it involves legal things.

If we agree on how to do it, I can do the modifications.

Sorry not to be aware at each time, family business is a full time, moreover with three boys ;-)

Let me know on the PR how you want to go forward

**Comment by larspalo** [comment]
Are we ready for a release or are we going to work on more issues first? Like getting the GO icon?

**Comment by oleg68** [comment]
> Are we ready for a release or are we going to work on more issues first? Like getting the GO icon?

No more issues for this release. You may create a PR with a CHANGELOG.md containing the release header 3.2.0-1. After merging this PR  I will create a release tag.

**Reply by oleg68** [reply]
3.2.0-1 is here. Please link issues for the next one.

**Comment by oleg68** [comment]
I'd get #711 for the first time.

@rousseldenis you have already started with ASIO in #718. Could you continue this work in #739?

**Reply by larspalo** [reply]
I think I'll do #735 even if it's not critical, but since I've looked into it anyway. Should we have a separate post with a list like before?

**Comment by oleg68** [comment]
> Should we have a separate post with a list like before?

Yes, it should be a place with checkboxes.

May be to try using Project feature in Github. I've never used it before.

**Comment by oleg68** [comment]
# 3.2.1

- [x] #741 @oleg68 
- [x] #720 @oleg68 
- [x] #711 @oleg68 
- [x] #8 @oleg68 
- [x] #735 @larspalo

**Comment by oleg68** [comment]
I'd also add #739 but @rousseldenis hasn't confirm that he would complete it

What other issues to take into the release?

**Comment by oleg68** [comment]
@larspalo @rousseldenis 

1. do you agree with this list or you want to modify it?
2. Should we increase the minor version after #741 (3.3.0 instead of 3.2.1)?

**Reply by rousseldenis** [reply]
Ok for 1. IMHO, upgrading major or minor version should correspond to a great amount of functional features or breaking changes. That one is a little aside of software itself, no?

**Reply by oleg68** [reply]
Is capability of double-click installing is a functional feature or not?

**Reply by rousseldenis** [reply]
It's more related to software install or deployment.

But we can consider it enhance software experience, so, yes, we can go for 3.3.0 after that.

**Comment by oleg68** [comment]
@larspalo @rousseldenis What will be the nearest release number: 3.2.1 or 3.3.0?

Do

- Adding polish translation
- Changed linux packaging

need to trigger increasing the minor version or not?

**Reply by oleg68** [reply]
Because everything is ready for the release, we must chose the version number now: 3.2.1 or 3.3.0.

@larspalo @rousseldenis please give yours suggestions.

**Reply by rousseldenis** [reply]
We can go for 3.3.0 in order to reflect new packaging.

On a side thinking, I suggest to define next milestone right after one release and use github ones as it allows to have a clear view of what issues should be adressed in it (and it allows to prepare even several versions in advance).

Example here for python pip dev :

![image](https://user-images.githubusercontent.com/19529533/136603413-e2b833ea-c72a-46f2-bb65-b4a2dbb1d477.png)

![image](https://user-images.githubusercontent.com/19529533/136603462-9a4a9753-d422-4463-bde1-ca9e834b3299.png)


What do you think about ?

**Comment by oleg68** [comment]
@rousseldenis Yes, I have created https://github.com/GrandOrgue/grandorgue/milestone/2 for the next release.

First of all I'd like to do #762.

Next I'd like to continue working with #28 and to do #763. Maybe it will be enouth for #28. Otherwise I'll create new issues necessary for #28.

@larspalo @rousseldenis please add your issues to the milestone.

**Comment by oleg68** [comment]
Because the new release is coming, I've started to collect issues for the next ones: https://github.com/GrandOrgue/grandorgue/milestone/3

@larspalo @rousseldenis 

Please share your ideas.

**Comment by oleg68** [comment]
@rousseldenis There are 4 PRs waiting for your approval: #810, #811, #814, #815.

Could you approve them?

**Reply by oleg68** [reply]
@larspalo @rousseldenis 

Do you have any ideas for the next release?

I'd like to take #701 and #847

May be it's a time to make some next steps with the application icon?

**Reply by larspalo** [reply]
I've been working quite a bit on LoopAuditioneer lately, and I'm a bit torn about what to focus on for GO. Eventually I'd like to look into #717 but that's maybe for a later release. For now the #24 could perhaps be a more pressing matter, as well as starting to update my screencasts on using GO (at https://www.youtube.com/playlist?list=PLNz9AJhjubTdoQoBf6TGADcTSOEyk0X5I) that serve as introduction tutorials for new GO users. Migrating the Sampleset page could also be included in #24 as it's referenced there.

**Comment by oleg68** [comment]
I'd like to vote for #24.

Unfortunally, I don't know how to do it.

**Reply by larspalo** [reply]
I'm reading on it right now. It seems to be fairly straightforward. A lot of the old information from Sourceforge needs updating though.

**Comment by oleg68** [comment]
> I'm reading on it right now. It seems to be fairly straightforward. 

May you assign #24 to youself?

> A lot of the old information from Sourceforge needs updating though.

May we do it in two steps:
1. Copy from sourceforge 1:1
2. Actualize it.

I think (1) would be sufficient for the next release.

**Reply by larspalo** [reply]
Yes, I'll assign #24 to me. I'll try and see how the wiki format of Sourceforge matches the input format here on GitHub. It might be so that every added wiki page will be like a commit so that you'll be notified about it, we'll see.

**Reply by Jean-Gabriel1982** [reply]
Hello all,

I have been very busy these days and now I will try to "come back" :-)

For my part, I think that it would be necessary to integrate new functionalities which are visible by the user.

there may indeed be the change of icon # 734, but also the reorganization of menus # 686, or the automatic closing of an organ when another # 707 is loaded (which can be very practical when plays a ceremony) ..

I know it's a lot of work and I take this message to sincerely thank @larspalo, @oleg68, @rousseldenis for the quality of their work and the reactivity to make GO evolve

@larspalo you suggested that I learn the code to help, but think about it, I don't have the time to get started right now and would be more of a burden than a help 😔

**Comment by oleg68** [comment]
@Jean-Gabriel1982 In my personal opinion usualy bugs should be fixed before adding new features or enchancements.

I'd like to add #707 because it looks like a bug (lacks of releasing memory somewhere). But I'm afraid that fixing #701 and #847 will take too much time for adding #707 to the same release. So I'd take #701 and #847 into the nearest release and #707 to the next one.

**Reply by larspalo** [reply]
Well, I'm not so sure that #707 is valid at all. GO is supposed to be able to be multi instanced - allowing having multiple organs open at the same time. It's the users responsibility to close an opened organ that's no longer wanted. This is all by design, not by lack of understanding!

**Reply by oleg68** [reply]
Answered in #707

**Comment by oleg68** [comment]
@larspalo @rousseldenis because 3.4.3 is comming it is the time for planning the next release.

The only bug rest is #707. I need some debug info from the author for fixing it.

Now I want to start developing #703. It is quite complex issue with user enchancements, so it will lead to increase the minor version up to 3.5.0.

**Comment by oleg68** [comment]
For now, there are two releases planned: 
- a small one `3.4.4` with #707
- a large one `3.5.0` with #703

@larspalo @rousseldenis do you see another issues to include?

@rousseldenis what is the current status of asio support now?

**Reply by larspalo** [reply]
I've been busy with a lot of other stuff lately. But I've started planning for an application for odf production. Initially as a standalone executable. While the integration with GO would be highy desirable, it's also problematic. Initially at least, I think the wannabe sampleset producers for GO would be quite satisfied with an application that could easily produce the odf for them with more or less all the bells and whistles even if it won't be directly inside of GO.

Besides this I've been thinking about #700 issue. Ideally I'd like a more detailed possibility of management of the settings and cache files, but as the last post of the issue starter of #700 suggest, it might be enough, at least to begin with, to provide a scan of the settings and cache files directories for matching files when an organ is deleted and ask if the user want to delete related settings and cache files too. Maybe an option to scan for possible removal of orphaned settings/cache files too...

**Reply by sandstone42** [reply]
I've been using Vincent Forman's Organ Builder program to get a basic ODF.

**Reply by larspalo** [reply]
> I've been using Vincent Forman's Organ Builder program to get a basic ODF.

Yes, that's certainly possible, but it's only available natively for windows (can be run under wine) and it also does some things wrong not to mention that it is incomplete.

I've previously written [jMakeOdf](https://github.com/larspalo/grandorgue-odf/tree/master/java/jMakeOdf) that's multiplatform and produce a resonably good odf skeleton. The problem with [jMakeOdf](https://github.com/larspalo/grandorgue-odf/tree/master/java/jMakeOdf) is that it's a command line application with the requirement of writing a config file for it to know how to parse a specific sampleset tree. This seems to be above what most users can easily figure out how to do. Thus my goal now is a GUI program with the same c++/wxwidgets language combo that GO has that can do more or less everything needed in an odf except that which already can be done from within GO itself.

**Reply by Jean-Gabriel1982** [reply]
All this sounds great !!  

What would be good indeed is that the user could by dragging and dropping, create his own ODF file.

He could thus create his organ composition, from stops to accessories (trembling, couplings, titrasses, etc.) and also create his right and left panels.  

We could imagine that the user (or the program) could act on very basic harmonization tasks, such as unifying the sound level in the case where the user imports sound samples from several organs.

PS maybe could we open a new topic for this discussion ?

**Comment by oleg68** [comment]
> But I've started planning for an application for odf production.

Sounds good. Making an ODF manually is quite difficult.

> Besides this I've been thinking about #700 issue. Ideally I'd like a more detailed possibility of management of the settings and cache files, but as the last post of the issue starter of #700 suggest, it might be enough, at least to begin with, to provide a scan of the settings and cache files directories for matching files when an organ is deleted and ask if the user want to delete related settings and cache files too. Maybe an option to scan for possible removal of orphaned settings/cache files too...

Are you going to get #700 for the nearest releases?

**Reply by larspalo** [reply]
> Are you going to get #700 for the nearest releases?

I won't make any promise on that. Better not count on it. I'll see when I have something that you can look at.

**Comment by Jean-Gabriel1982** [comment]
> But I've started planning for an application for odf production.

Wahou that's a very good news !!
